scripts/all/setup-env.sh

```diff
@@ -0,0 +1,74 @@
+### Load environment modules here
+### Assign names as relevant
+
+mygcc="gcc/9.4.0"
+myclang="clang/13.0.0"
+mycuda="cuda/11.4.0"
+myrocm="rocm"
+mycmake="cmake"
+mympi="openmpi"
+
+if [ "$1" = "hpc" ]
+then
+    module purge
+    if [ "$2" = "cuda" ]
+    then
+        module purge
+        module load ${mygcc}
+        module load ${mycuda}
+    elif [ "$2" = "hip" ]
+    then
+        module purge
+        module load ${myclang}
+        module load ${myrocm}
+    else
+        module load ${mygcc}
+    fi
+    module load ${mycmake}
+    module load ${mympi}
+    module -t list
+fi
+
+
+my_device="$2"
+
+my_build="build-all"
+if [ -z $3 ]
+then
+    my_build="${my_build}-${my_device}"
+else
+    my_build=$3
+fi
+
+export scriptdir=`pwd`
+
+cd ../..
+export topdir=`pwd`
+export basedir=${topdir}
+export srcdir=${basedir}/src
+export libdir=${topdir}/lib
+export matardir=${libdir}/Elements/matar
+export trilinosdir=${libdir}
+export builddir=${basedir}/${my_build}
+#export installdir=${basedir}/install-kokkos/install-kokkos-${my_device}
+
+export FIERRO_BASE_DIR=${basedir}
+export FIERRO_SOURCE_DIR=${srcdir}
+export FIERRO_EXPLICIT_SOURCE_DIR=${srcdir}/Parallel-Solvers/Parallel-Explicit
+export FIERRO_IMPLICIT_SOURCE_DIR=${srcdir}/Parallel-Solvers/Implicit-Lagrange
+export FIERRO_BUILD_DIR=${builddir}
+
+#export KOKKOS_SOURCE_DIR=${matardir}/src/Kokkos/kokkos
+#export KOKKOS_BUILD_DIR=${builddir}/kokkos
+#export KOKKOS_INSTALL_DIR=${installdir}/kokkos
+
+# Do this differently (in src tree) than other libs because
+# of compile time
+export TRILINOS_SOURCE_DIR=${trilinosdir}/Trilinos
+export TRILINOS_BUILD_DIR=${TRILINOS_SOURCE_DIR}/build-${my_device}
```

Comment:
Trilinos build directories actually depend more on the parallelization than hpc vs macos -- especially since you wouldn't have both hcp and macos on the same machine, presumably. 

So we should append the parallelization instead of "my_device"

Reply:
Same thing for the fierro build dir btw

Reply:
I can change the name of that variable. I don't know why I called it 'my_device' when that is actually referring to the parallelism

Reply:
Ahh woops I see that now. Yea ok maybe a different variable name then xD.